### PR TITLE
fix: #23

### LINKED ISSUE #23: Implement Sorting for List
Implement a sort method that sorts elements of a `List` structure based on a comparison function using quicksort sorting algorithm.

Points to be noted:

- The `sort` method should be compatible with any list type and not limited to lists of numbers or strings. Use generics to achieve this.

- The `sort` method should accept three parameters: 
    - `list`, which is the list to be sorted of type `List[T]`. 
    -  `cmp`, which is a comparison function with signature `func(elem1 T, elem2) int` and should return -1.0 or 1 depending on the result of the comparison of the two elements.
    
If you'd like to work on this, please comment "picking this up" below, and I'll assign the issue to you.

Please take a look at our [CONTRIBUTING.md](https://github.com/gopher-utils/go-collections/blob/main/CONTRIBUTING.md) before you start working on this.

### PR DESCRIPTION
**Pull request by DGclasher**
### Description

Added implementation of list sorting

### Related Issues

Closes #23

### CODE REVIEW

**Comment by aswath-s-tw** [comment]
@DGclasher Can you also add unit tests for your changes and ensure coverage ?

File: list/list.go [hunk 1]
```diff
@@ -197,6 +197,42 @@ func Reduce[T collections.CollectionElement](l *List[T], callback listReduceFunc
 	return result
 }
 
+// Define a generic Sort method for the List structure.
```
Comment (aswath-s-tw) [inline list/list.go]: Can you make this comment more descriptive ? 
Ref: https://github.com/gopher-utils/go-collections/blob/1d35847621b6eb1e6ef8ba38c9217ad7aa56ab5c/list/list.go#L98

Comment (DGclasher) [reply list/list.go]: alright, working on that

**Comment by DGclasher** [comment]
added tests

**Comment by aswath-s-tw** [comment]
@DGclasher please fix the errors !

**Pull request closed without merging**